PWGHF/HFC/TableProducer/correlatorLcHadrons.cxx

```diff
@@ -0,0 +1,700 @@
+      for (const auto& [t1, t2] : o2::soa::combinations(o2::soa::CombinationsFullIndexPolicy(tracks1, tracks2))) {
+        if (!TESTBIT(t1.hfflag(), aod::hf_cand_3prong::DecayType::LcToPKPi)) {
+          continue;
+        }
+        if (yCandMax >= 0. && std::abs(hfHelper.yLc(t1)) > yCandMax) {
+          continue;
+        }
+        // LcToPKPi and LcToPiKP division
+        if (t1.isSelLcToPKPi() >= selectionFlagLc) {
+          entryLcHadronPair(getDeltaPhi(t1.phi(), t2.phi()),
+                            t1.eta() - t2.eta(),
+                            t1.pt(),
+                            t2.pt(),
+                            poolBin);
+          entryLcHadronRecoInfo(hfHelper.invMassLcToPKPi(t1), false);
+        }
+        if (t1.isSelLcToPiKP() >= selectionFlagLc) {
+          entryLcHadronPair(getDeltaPhi(t1.phi(), t2.phi()),
+                            t1.eta() - t2.eta(),
+                            t1.pt(),
+                            t2.pt(),
+                            poolBin);
+          entryLcHadronRecoInfo(hfHelper.invMassLcToPiKP(t1), false);
+        }
+      }
+    }
+  }
+  PROCESS_SWITCH(HfCorrelatorLcHadrons, processDataMixedEvent, "Process Mixed Event Data", false);
+
+  void processMcRecMixedEvent(SelectedCollisions const& collisions,
+                              SelectedCandidatesMcRec const& candidates,
+                              SelectedTracks const& tracks)
+  {
+    auto tracksTuple = std::make_tuple(candidates, tracks);
+    Pair<SelectedCollisions, SelectedCandidatesMcRec, SelectedTracks, BinningType> pairMcRec{corrBinning, 5, -1, collisions, tracksTuple, &cache};
+
+    for (const auto& [c1, tracks1, c2, tracks2] : pairMcRec) {
+      int poolBin = corrBinning.getBin(std::make_tuple(c2.posZ(), c2.multFV0M()));
+      for (const auto& [t1, t2] : o2::soa::combinations(o2::soa::CombinationsFullIndexPolicy(tracks1, tracks2))) {
+        if (yCandMax >= 0. && std::abs(hfHelper.yLc(t1)) > yCandMax) {
+          continue;
+        }
+        if (t1.isSelLcToPKPi() >= selectionFlagLc) {
+          entryLcHadronPair(getDeltaPhi(t1.phi(), t2.phi()),
+                            t1.eta() - t2.eta(),
+                            t1.pt(),
+                            t2.pt(),
+                            poolBin);
+          entryLcHadronRecoInfo(hfHelper.invMassLcToPKPi(t1), false);
+        }
+        if (t1.isSelLcToPiKP() >= selectionFlagLc) {
+          entryLcHadronPair(getDeltaPhi(t1.phi(), t2.phi()),
+                            t1.eta() - t2.eta(),
+                            t1.pt(),
+                            t2.pt(),
+                            poolBin);
+          entryLcHadronRecoInfo(hfHelper.invMassLcToPiKP(t1), false);
+        }
+      }
+    }
+  }
+  PROCESS_SWITCH(HfCorrelatorLcHadrons, processMcRecMixedEvent, "Process Mixed Event McRec", false);
+
+  void processMcGenMixedEvent(SelectedCollisionsMcGen const& collisions,
```

Comment:
What's the reason to keep MC gen and MC reco separated in two process functions? Do you need only one of the two in some cases?